packages/calypso-e2e/src/lib/blocks/audio-block.ts

```diff
@@ -0,0 +1,50 @@
+import { Page, ElementHandle } from 'playwright';
+
+const selectors = {
+	block: '.wp-block-audio',
+	fileInput: '.components-form-file-upload input[type="file"]',
+	spinner: '.components-spinner',
+};
+
+/**
+ * Represents the Audio block.
+ */
+export class AudioBlock {
+	static blockName = 'Audio';
+	block: ElementHandle;
+
+	/**
+	 * Constructs an instance of this block.
+	 *
+	 * @param {ElementHandle} block Handle referencing the block as inserted on the Gutenberg editor.
+	 */
+	constructor( block: ElementHandle ) {
+		this.block = block;
+	}
+
+	/**
+	 * Uplaods the target file at the supplied path to WPCOM.
+	 *
+	 * @param {string} path Path to the file on disk.
+	 */
+	async upload( path: string ): Promise< void > {
+		// The `input` element has a display:none style set and it cannot be found normally even if
+		// waitForSelector({state: 'hidden'}) is used.
+		const input = ( await this.block.$( selectors.fileInput ) ) as ElementHandle;
+		await input?.setInputFiles( path );
+		await Promise.all( [
+			this.block.waitForSelector( selectors.spinner, { state: 'hidden' } ),
+			this.block.waitForElementState( 'stable' ),
```

Comment:
The block has a spinner overlay while the image is uploading, which is replaced with the image itself once the upload is complete. During this time, the bounding box of the block changes.